src/averageAges.js

```diff
@@ -15,6 +15,18 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
+  const men = !century
+    ? people.filter(person => person.sex === 'm')
+    : people.filter(person => person.sex === 'm'
+      && Math.ceil(person.died / 100) === century);
+
+  const menAge = men.map(man => man.died - man.born);
+
+  const averageAge = menAge.reduce((prev, curr) => prev + curr, 0)
+    / menAge.length;
+
+  return averageAge;
```

Comment:
move logic related to calculate average age to the separate function so you will be able to reuse it in yours main functions